reducers/index.js

```diff
@@ -37,6 +39,16 @@ module.exports = function navigate(state, {type, payload}) {
         stack: stack.splice(index, 1, payload),
       });
 
+    case REPLACE_AT_INDEX:
+      invariant(
+        payload.index < stack.count() - 1 && payload.index > 0,
+        'Index out of bounds when trying to replaceAtIndex',
```

Comment:
Sometimes it's handy to know which values you actually have. So, for instance, this message can be enhanced:
```
Index out of bounds when trying to replaceAtIndex: your navigation stack contains ${stack.count()} elements, but you're trying to replace an element with index ${payload.index}
```